core/src/main/scala/com/lightbend/paradox/markdown/Url.scala

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright © 2015 - 2016 Lightbend, Inc. <http://www.lightbend.com>
+ * 
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * 
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.lightbend.paradox.markdown
+
+import java.net.{ URI, URISyntaxException }
+
+/**
+ * Small wrapper around URI to help update individual components.
+ */
+case class Url(base: URI) {
+  def withEndingSlash = base.getPath match {
+    case path if path.endsWith("/index.html") => this
+    case path                                 => copy(path = path + "/")
```

Comment:
The main goal here is to avoid links that will automatically redirect, yet allow to spell out with `.../index.html` if so desired.